docs-en/chapter_computational_complexity/summary.md

```diff
@@ -2,48 +2,48 @@
+- Time complexity is used to measure the trend of the algorithm's running time with data volume growth, which can effectively evaluate the algorithm's efficiency. However, it may fail in some cases, such as when the input volume is small or the time complexities are similar, making it hard to compare the algorithm's efficiency accurately.
+- The worst time complexity is denoted by big $O$ notation, which corresponds to the asymptotic upper bound of the function, reflecting the growth rate in the number of operations $T(n)$ as $n$ tends to positive infinity.
+- The time complexity can be determined in two steps: first, count the number of operations, then select the asymptotic upper bound.
+- Common time complexities, from lowest to highest, are $O(1)$, $O(\log n)$, $O(n)$, $O(n \log n)$, $O(n^2)$, $O(2^n)$, and $O(n!)$.
+- The time complexity of certain algorithms is not fixed and depends on the distribution of the input data. The time complexity can be categorized into worst, best, and average. The best time complexity is not commonly achieved because the input data must meet strict conditions to achieve the best case.
+- The average time complexity reflects the efficiency of an algorithm with random data inputs, which is closest to the performance of algorithms in real-world scenarios. Calculating the average time complexity requires statistical analysis of input data and a synthesized mathematical expectation.
 
 **space complexity**
 
-- Space complexity acts similarly to time complexity and is used to measure the tendency of an algorithm's space occupation to grow with the amount of data.
-- The memory space associated with the operation of an algorithm can be categorized into input space, staging space, and output space. Normally, the input space is not counted in the space complexity calculation. The transient space can be divided into instruction space, data space, and stack frame space, of which the stack frame space usually affects the space complexity only in the recursion function.
-- We usually focus only on the worst-case space complexity, i.e., the space complexity of the statistical algorithm with the worst-case input data and the worst-case running time complexity.
-- Common space complexities are $O(1)$, $O(\log n)$, $O(n)$, $O(n^2)$ and $O(2^n)$ in descending order.
+- Space complexity serves a similar purpose to time complexity and is used to measure the trend of space occupied by an algorithm as the data volume increases.
+- The memory space associated with the operation of an algorithm can be categorized into input space, temporary storage space, and output space. Normally, the input space is not considered when determining space complexity. The transient space can be classified into instruction space, data space, and stack frame space, and the stack frame space usually only affects the space complexity for recursion functions.
```

Comment:
We may need to use the term universally here. However, `temporary storage space` and `transient space` are mixed in this paragraph. I would prefer using `temporary space` which is more succinct.

When I search **space complexity**, most of articles, including Wikipedia, group by `input space` and `auxiliary space`. I was wondering which term serves best in this context.

Reply:
Thanks for the insight. For this one, I will update to _temporary space_ for now, since _auxiliary space_ is little confused here since we have an output space.